nncf/quantization/algorithms/min_max/torch_backend.py

```diff
@@ -0,0 +1,231 @@
+"""
+ Copyright (c) 2023 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Dict, List, Tuple
+
+from nncf.common.hardware.config import HWConfig
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.definitions import NNCFGraphNodeType
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.operator_metatypes import OperatorMetatype
+from nncf.common.graph.model_transformer import ModelTransformer
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.layer_attributes import WeightedLayerAttributes
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.quantization.initialization.range import RangeInitConfig
+from nncf.common.tensor_statistics.statistics import MinMaxTensorStatistic
+from nncf.common.utils.backend import BackendType
+from nncf.common.graph.transformations.commands import TransformationPriority
+
+from nncf.torch.hardware.config import PTHWConfig
+from nncf.torch.quantization.default_quantization import DEFAULT_PT_QUANT_TRAIT_TO_OP_DICT
+from nncf.torch.quantization.layers import QUANTIZATION_MODULES
+from nncf.torch.quantization.layers import PTQuantizerSpec
+from nncf.torch.quantization.layers import get_scale_shape
+from nncf.torch.quantization.init_range import PTRangeInitCollectorParams
+from nncf.torch.quantization.init_range import StatCollectorGenerator
+from nncf.torch.nncf_network import PTModelTransformer
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.graph.graph import PTTargetPoint
+from nncf.torch.graph.transformations.commands import PTInsertionCommand
+from nncf.torch.tensor_statistics.collectors import PTMinMaxStatisticCollector
+from nncf.torch.tensor_statistics.statistics import PTMinMaxTensorStatistic
+from nncf.torch.tensor_statistics.collectors import PTMeanMinMaxStatisticCollector
+
+from nncf.quantization.algorithms.min_max.backend import MinMaxAlgoBackend
+from nncf.quantization.algorithms.min_max.backend import ALGO_BACKENDS
+
+import nncf.torch.graph.operator_metatypes as om
+
+@ALGO_BACKENDS.register(BackendType.TORCH)
+class PTMinMaxAlgoBackend(MinMaxAlgoBackend):
+
+    @property
+    def layers_with_weights_metatypes(self) -> List[OperatorMetatype]:
+        return [
+            om.PTModuleConv1dMetatype,
```

Comment:
Do we really support 1D-2D convolutions?

Reply:
Pytorch should support 1D convolutions as torch quantization algorithm supports them. All code of statistic collectors and quantizers builder are taken from original Torch quantization algoritm

Reply:
For this purpose, It might be useful to add some tests for such types of layers. But not in this PR, I suppose.